RyuSocks/Auth/Packets/UsernameAndPasswordRequest.cs

```diff
@@ -0,0 +1,157 @@
+            }
+        }
+
+        public string Password
+        {
+            get
+            {
+                return Encoding.ASCII.GetString(Bytes.AsSpan(3 + UsernameLength, PasswordLength));
+            }
+            set
+            {
+                ArgumentOutOfRangeException.ThrowIfGreaterThan(value.Length, 0xFF);
+                PasswordLength = (byte)value.Length;
+                Encoding.ASCII.GetBytes(value).CopyTo(Bytes.AsSpan(3 + UsernameLength, PasswordLength));
+            }
+        }
+
+        public UsernameAndPasswordRequest(byte[] packetBytes)
+        {
+            if (packetBytes.Length is < 4 or > 513)
+            {
+                throw new InvalidOperationException($"IncomingPacket is of wrong length: Expected: > 4 || < 513; Actual {packetBytes.Length}");
+            }
+
+            Bytes = packetBytes;
+            Version = Bytes[0];
+            UsernameLength = Bytes[1];
+
+            try
+            {
+                Username = Encoding.ASCII.GetString(Bytes[2..(2 + UsernameLength)]);
+            }
+            catch (Exception e)
+            {
+                throw new InvalidOperationException($"UsernameLength and actual Username length do not match: {Bytes[1]:X} (throws {e})");
+            }
+
+            PasswordLength = Bytes[2 + UsernameLength];
+
+            try
+            {
+                Password = Encoding.ASCII.GetString(Bytes[(2 + UsernameLength + 1)..((2 + UsernameLength + 1) + PasswordLength)]);
+            }
+            catch (Exception e)
+            {
+                throw new InvalidOperationException($"PasswordLength and actual Password length do not match: {Bytes[1]:X} (throws {e})");
+            }
+        }
+
+        public UsernameAndPasswordRequest(string username, string password)
+        {
+            Bytes = new byte[4 + username.Length + password.Length];
+            Version = AuthConsts.UsernameAndPasswordVersion;
+            UsernameLength = (byte)username.Length;
+            Username = username;
+            PasswordLength = (byte)password.Length;
+            Password = password;
+        }
+
+        public override void Validate()
+        {
+            const int MinLength = 4;
+            const int MaxLength = 513;
+
+            if (Bytes.Length is < MinLength or > MaxLength)
+            {
+                throw new InvalidOperationException($"$Package has wrong Length: {Bytes.Length:X} (Expected: >= {MinLength} || <= {MaxLength})");
+            }
+
+            if (Version != AuthConsts.UsernameAndPasswordVersion)
+            {
+                throw new InvalidOperationException($"${nameof(Version)} is invalid: {Version:X} (Expected: {AuthConsts.UsernameAndPasswordVersion:X})");
+            }
+
+            if (Username == null)
+            {
+                throw new InvalidOperationException($"$Username is invalid: {Username} (Expected: Literally Anything but null)");
+            }
+
+            if (Password == null)
```

Comment:
```suggestion
            if (string.IsNullOrEmpty(Password))
```

Password could also be an empty string.

Reply:
Not addressed yet.